crates/nu-std/CONTRIBUTING.md

```diff
@@ -0,0 +1,191 @@
+If your idea isn't quite so broadly applicable, 
+consider publishing it in [nu_scripts](https://github.com/nushell/nu_scripts).
+
+Preliminary discussions should focus on the *user benefit* your idea would provide.  
+How many users will be affected by your idea, how would (a good implementation of) it 
+improve their Nushell experience?
+Given consensus on the user benefit, the team will be motivated to 
+help you create, deploy and maintain a solution long term.
+
+## Lifecycle of a change
+
+1. Verify the team thinks your idea is potentially relevant and useful, as above.
+1. If it's more than a simple bug fix, open a placeholder PR 
+as soon as you get started and mark it as work in progress (WIP).
+This will alert other contributors that you're working in this area and 
+advertise roughly what scope of changes you're thinking of.
+See [below](the_pr) for details.
+1. Get things working in your local development environment.  
+If you have questions along the way, you can post a question in your PR 
+or have a more casual discussion with Nushell fans on 
+[Discord implementation-chat channel](https://discord.com/channels/601130461678272522/615962413203718156)
+1. When you get to an appropriate state of doneness, push your changes to the PR and remove the WIP flag.
+2. Team members and other contributors will then review your PR.  
+Respond to any review comments they raise and resolve them one way or another.
+(Not all comments demand you make a change!)
+1. When you and the team are comfortable with the PR, 
+a team member will merge it into the repo and you can delete your working branch.
+2. If you've added a whole new command or made a breaking change, 
+(strongly) consider writing up a section for the release notes.  
+Currently, release notes are maintained in a different repo, [nushell.github.io](https://github.com/nushell/nushell.github.io). 
+Make your change in a local clone of that repo and submit a PR to the release notes repo to get it integrated.
+
+## Developing
+
+### Setup
+To work on the standard library, you won't necessarily need to compile or debug the interpreter itself, 
+so you might not need to install the Rust toolchain. 
+But if you want to do that, see [nushell CONTRIBUTING.md](https://github.com/nushell/nushell/blob/main/CONTRIBUTING.md).
+
+1. Clone the Nushell repo containing the standard library.  
+Currently, that's the [Nushell interpreter source repo](https://github.com/nushell/nushell).
+    ```
+    git clone https://github.com/nushell/nushell
+    ```
+
+2. Arrange a runnable Nushell interpreter  
+  Either install a [pre-built distribution](https://github.com/nushell/nushell/releases), or build from source:
+    ```
+    git clone https://github.com/nushell/nushell
+    cd nushell
+    cargo run
+    ```
+1. Focus your IDE on the standard library folder  
+(Currently, it is a Rust crate within the Nushell repo)
+
+    ```
+    cd /path/to/nushell/repo
+    code .  # opinionated: we love vscode!
```

Comment:
booooh we love `vim` :stuck_out_tongue_closed_eyes:

Reply:
I knew it!  (where are the emacs bigots?) 
I'll fuzzy this up...

Reply:
i've not seen Emacs pop up in the `nushell` community for now :smirk: